### PR TITLE
EP Merge: add checks for get EP400 claim details

### PR DESCRIPTION
**Pull request by dfitchett**
## What was the problem?

Due to issue discovered by NWQ outlined in this [ticket](https://github.com/department-of-veterans-affairs/abd-vro/issues/2734), EP Merge App needs to be updated to check if the supplemental EP has an end product code in the range of 400-409 and a benefit claim type code (aka claim label) of `400SUPP`.

Associated tickets or Slack threads:
- #2734 
- Stacked on top of PR:
  - #2746 

## How does this fix it?[^1]

When processing a merge request, after getting the pending EP claim details, the process now includes getting the EP400 claim details and checks if the end product code and benefit claim type label are valid. 

## How to test this PR
- pull code
- Unit Test:
  - from /domain-ee/ee-ep-merge-app run pytest
- Integration Tests:
  - run [base platform](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Docker-Compose#platform-base)
  - from /domain-ee/ee-ep-merge-app run pytest ./integration
- End to End tests:
  - run this [CI](https://github.com/department-of-veterans-affairs/abd-vro/actions/workflows/ee-ep-merge-end-to-end.yml) action to verify successful integration test
  - run end to end tests locally:
    - `export BIP_CLAIM_URL=mock-bip-claims-api:20300`
    - run bgs and bip services and mocks:
      - `COMPOSE_PROFILES="bip,bgs" ./gradlew :app:dockerComposeUp`
      - `COMPOSE_PROFILES="bip,bgs" ./gradlew -p mocks :dockerComposeUp`
    - `./gradlew :domain-ee:ee-ep-merge-app:endToEndTest`

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

**Review by msnwatson — APPROVED** [review]

**Review by nelsestu — APPROVED** [review]
I see the additional benefitsClaimType handling being added to the processing of EP 400 requests.  Based on the description of the change needed in the PR, I can concur that the changes appear to apply the desired behavior.

File: domain-ee/ee-ep-merge-app/src/python_src/service/ep_merge_machine.py [hunk 1]
```diff
@@ -40,6 +40,9 @@
 JOB_SKIPPED_MERGE_METRIC = 'job.skipped_merge'
 JOB_NEW_CONTENTIONS_METRIC = 'job.new_contentions'
 
+EP400_PRODUCT_CODES = frozenset([str(i) for i in range(400, 410)])
+EP400_BENEFIT_CLAIM_TYPE_CODES = frozenset(['400SUPP'])
```
Comment (nanotone) [inline domain-ee/ee-ep-merge-app/src/python_src/service/ep_merge_machine.py:44]: I like this. If you want to be even more clever, you can just do `EP400_PRODUCT_CODES = range(400, 410)` -- range objects are magical lightweight stand-ins for their more explicit counterparts.

**Review by nanotone — APPROVED** [review]
Looks great! Just the one suggestion, but this is approved.

File: domain-ee/ee-ep-merge-app/src/python_src/service/ep_merge_machine.py (hunk 1 above)
Comment (dfitchett) [reply domain-ee/ee-ep-merge-app/src/python_src/service/ep_merge_machine.py:44]: Ah nice, but need it to be strings!

Comment (nanotone) [reply domain-ee/ee-ep-merge-app/src/python_src/service/ep_merge_machine.py:44]: Ahaha, of course. 👍

**Pull request merged**